src/part-guide/concurrency.md

```diff
@@ -31,61 +31,61 @@ Even where IO[^io-def] happens (printing using `println!` is IO - it is outputti
+Let's look at IO again. What happens when a thread blocks waiting for IO? In a system with threads, the OS will pause the thread (it's just going to be waiting in any case) and wake it up again when the IO is complete[^busywait]. Depending on the scheduling algorithm, it might take some time after the IO completes until the OS wakes up the thread waiting for IO, since the OS might wait for other threads to get some work done. So now things are much more efficient: while one thread waits for IO, another thread (or more likely, many threads due to multitasking) can make progress. But, from the perspective of the thread doing IO, things are still sequential - it waits for the IO to finish before starting the next operation.
 
-A thread can also choose to pause itself by calling a `sleep` function, usually with a timeout. In this case the OS pauses the thread at the threads own request. Similar to pausing due to pre-emption or IO, the OS will wake the thread up again later (after the timeout) to continue execution.
+A thread can also choose to pause itself by calling a [`sleep`](https://doc.rust-lang.org/stable/std/thread/fn.sleep.html) function, usually with a timeout. In this case the OS pauses the thread at the thread's own request. Similar to pausing due to pre-emption or IO, the OS will wake the thread up again later (after the timeout) to continue execution.
 
-When an OS pauses one thread and starts another (for any reason), it is called *context switching*. The context being switched includes the registers, operating system records, and the contents of many caches. That's a non-trivial amount of work. Together with the transfer of control to the OS and back to a thread, and the costs of working with stale caches, context switching is an expensive operation.
+When an OS pauses one thread and starts another (for any reason), it is called *context switching*. The context being switched includes the registers, operating system records, and the contents of many caches. That's a non-trivial amount of work. On top of that, there is a transfer of control to the OS and back to the thread and the costs of working with stale caches. Adding all these costs together makes context switching an expensive operation.
 
-Finally, note that some hardware or OSs do not support processes or threads, this is more likely in the embedded world.
+Finally, note that some hardware or OSs do not support processes or threads. In the embedded world, there is no operating system, so there is nothing to provide processes or threads.[^rtos]
```

Comment:
I prefer the first formulation here since this is not meant to be specific to Rust and as you note some embedded OSs do have threads.